### DISCUSSION TITLE
Explain removal of environment variables `DEFAULT_EMAIL` and `DEFAULT_PASSWORD`

### DISCUSSION THREAD

**Discussion by ToshY** [mealie-recipes/mealie#3242] [Q&A] [answered]
The docs are somewhat contradicting regarding docker compose setup with environment variables; the "installation checklist" states that you need to set [`DEFAULT_EMAIL`](https://docs.mealie.io/documentation/getting-started/installation/installation-checklist/#step-2-customizing-the-docker-composeyaml-files) but at same time all examples are missing this environment variable.

After looking through the history, I saw `DEFAULT_PASSWORD` was mentioned/added in the docs in #2691 , but PR #2722 shows `DEFAULT_EMAIL` and `DEFAULT_PASSWORD` were removed entirely.

The reason in #2722  states:

> Probably 5 times a week someone asks what the default credentials are in some for or another. This PR resolves this issue by doing the following
> 1. Removes the DEFAULT_EMAIL and DEFAULT_PASSWORD variables, or at least changes the name and removes them from the documentation.

I've passed those environment variables to the container, but that didn't do anything, so I guess they are completely removed now, which brings me to my question: Isn't it possible to allow custom email/password for initial setup and just fallback to default email/password if those environment variables are empty?

I don't understand why the functionality was removed completely as the ability to supply your own email/password makes it easier (and more secure) to setup the container without manual intervention.

---

Docker: `v1.2.0`
Tag: 42a33cd993b4638f181d4083a7c5d21bf110e4e4

**Comment by boc-the-git** [comment, marked as answer]
> the "installation checklist" states that you need to set [DEFAULT_EMAIL](https://docs.mealie.io/documentation/getting-started/installation/installation-checklist/#step-2-customizing-the-docker-composeyaml-files) but at same time all examples are missing this environment variable.

As you've noted, that environment variable is no longer in use. PR's are always welcome (and appreciated!) to improve the docs.

> Isn't it possible to allow custom email/password for initial setup and just fallback to default email/password if those environment variables are empty?

Yes, that would be technically possible. It doesn't really improve the product in any meaningful way, and adds a very small complication into the initial setup. It's not worth it.

I don't expect us to bring back this functionality in the short term, without a particularly good reason for doing so.
